.scaffold/tests/bats/deployment--artifact.bats

```diff
@@ -0,0 +1,124 @@
+  pushd "${LOCAL_REPO_DIR}" >/dev/null || exit 1
+  setup_ssh_key_fixture
+
+  # Assert missing DREVOPS_SSH_PREFIX
+  run scripts/drevops/setup-ssh.sh
+  assert_failure
+  assert_output_contains "Missing the required DREVOPS_SSH_PREFIX environment variable"
+
+  # Assert using default SSH file
+  export DREVOPS_SSH_PREFIX="test"
+  local file=${HOME}/.ssh/id_rsa
+  run scripts/drevops/setup-ssh.sh
+  assert_failure
+  assert_output_contains "Using default SSH file ${file}."
+  assert_output_contains "SSH key file ${file} does not exist."
+
+  ssh-keygen -t rsa -b 4096 -N "" -f "${SSH_KEY_FIXTURE_DIR}/id_rsa"
+  export DREVOPS_SSH_PREFIX="test"
+  local file=${HOME}/.ssh/id_rsa
+  run scripts/drevops/setup-ssh.sh
+  assert_success
+  assert_output_contains "Using default SSH file ${file}."
+  assert_output_contains "Using SSH key file ${file}."
+
+  # Assert using fingerprint no ssh key
+  export DREVOPS_SSH_PREFIX="test"
+  export DREVOPS_test_SSH_FINGERPRINT="TEST"
+  run scripts/drevops/setup-ssh.sh
+  assert_failure
+  assert_output_contains "Found variable DREVOPS_test_SSH_FINGERPRINT with value ${DREVOPS_test_SSH_FINGERPRINT}."
+  assert_output_contains "Using fingerprint-based deploy key because fingerprint was provided."
+  assert_output_contains "SSH key file ${HOME}/.ssh/id_rsa_${DREVOPS_test_SSH_FINGERPRINT} does not exist."
+
+  # Assert using fingerprint with ssh key
+  export DREVOPS_SSH_PREFIX="test"
+  export DREVOPS_test_SSH_FINGERPRINT="TEST"
+  ssh-keygen -t rsa -b 4096 -N "" -f "${SSH_KEY_FIXTURE_DIR}/id_rsa_${DREVOPS_test_SSH_FINGERPRINT}"
+  local file="${SSH_KEY_FIXTURE_DIR}/id_rsa_${DREVOPS_test_SSH_FINGERPRINT}"
+  declare -a STEPS=(
+    "Found variable DREVOPS_test_SSH_FINGERPRINT with value ${DREVOPS_test_SSH_FINGERPRINT}."
+    "Using fingerprint-based deploy key because fingerprint was provided."
+    "Using SSH key file ${file}."
+    "@ssh-add -l # ${file}"
+    "SSH agent has ${file} key loaded."
+  )
+  mocks="$(run_steps "setup")"
+  run scripts/drevops/setup-ssh.sh
+  assert_success
+  run_steps "assert" "${mocks[@]}"
+
+  # Assert does not have key loaded
+  export DREVOPS_SSH_PREFIX="test"
+  export DREVOPS_test_SSH_FINGERPRINT="TEST"
+  local file="${SSH_KEY_FIXTURE_DIR}/id_rsa_${DREVOPS_test_SSH_FINGERPRINT}"
+  declare -a STEPS=(
+    "Found variable DREVOPS_test_SSH_FINGERPRINT with value ${DREVOPS_test_SSH_FINGERPRINT}."
+    "Using fingerprint-based deploy key because fingerprint was provided."
+    "Using SSH key file ${file}."
+    "@ssh-add -l # The agent has no identities."
+    "SSH agent does not have a required key loaded. Trying to load."
+    "- SSH agent has ${file} key loaded."
+    "@ssh-add -D"
+    "@ssh-add ${file}"
+    "@ssh-add -l # ${file}"
+    "Finished SSH setup."
+  )
+  mocks="$(run_steps "setup")"
+  run scripts/drevops/setup-ssh.sh
+  assert_success
+  run_steps "assert" "${mocks[@]}"
+
+
+  # Finding key with matching fingerprint.
+  export DREVOPS_SSH_PREFIX="test"
```

Comment:
@AlexSkrypnyk I think this is a bug by the way

Reply:
@AlexSkrypnyk Could you please review the style of this unit test - does this meet your requirements? And I will continue on the rest of the deployment unit tests if this meets what you had in mind

Reply:
@richardgaunt 
Thank you for working on this

We want to test every script in `scripts/drevops` dir
We then want to test all paths in each script, meaning that it would require multiple tests for the same script with different input parameters.

In this instance, where we are testing correct inclusion of the available SSH key, we would want to:
1. Cleanup current env variable values (use `unset VAR`)
2. Populate the env with test data: create valid SSH key (for positive test) or broken SSH key (for negative test) AND add the fingerprint in different formats.
3. Setup tests, which will mock commands used in the script itself (for example, `ssh-agent` does not need to be real and can be just a mock that returns a success code or  error code)
3. Call the script
4. Assert the output of the script by calling the test steps helper.